packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -166,10 +166,8 @@ const SceneView = ({
 
   const isParentHeaderShown = React.useContext(HeaderShownContext);
   const parentHeaderHeight = React.useContext(HeaderHeightContext);
-  const headerHeight = getDefaultHeaderHeight(
-    useSafeAreaFrame(),
-    false,
-    insets.top
+  const [headerHeight, setHeaderHeight] = React.useState(
+    getDefaultHeaderHeight(useSafeAreaFrame(), false, insets.top)
```

Comment:
I think we should do something like this:
```suggestion
  const defaultHeaderHeight = getDefaultHeaderHeight(
    useSafeAreaFrame(),
    false,
    insets.top
  );
    
  const [customHeaderHeight, setCustomHeaderHeight] = React.useState(defaultHeaderHeight)
  
  const headerHeight = header != null ? customHeaderHeight : defaultHeaderHeight;
```

This is so that we use correct height if custom header is removed later.